### PR TITLE
refactor: remove usage of async module

### PR DESCRIPTION
**Pull request by AVaksman**
Towards googleapis/google-cloud-node/issues/2883

- [x] Tests and linter pass

### CODE REVIEW

File: system-test/storage.ts [hunk 1]
```diff
@@ -15,11 +15,12 @@
  */
 
 import * as assert from 'assert';
-import * as async from 'async';
 import * as crypto from 'crypto';
 import * as fs from 'fs';
 import fetch from 'node-fetch';
 const normalizeNewline = require('normalize-newline');
+const pLimit = require('p-limit');
```
Comment (JustinBeckwith) [inline system-test/storage.ts]:
```suggestion
import pLimit = require('p-limit');
```

File: system-test/storage.ts [hunk 2]
```diff
@@ -15,11 +15,12 @@
...
+const {promisify} = require('util');
```
Comment (JustinBeckwith) [inline system-test/storage.ts]:
```suggestion
import {promisify} from 'util';
```

File: system-test/storage.ts [hunk 3]
```diff
@@ -946,14 +898,9 @@ describe('storage', () => {
...
-    after(done => {
-      async.series(
-        bucketsToCreate.map(bucket => {
-          return (done: DeleteBucketCallback) => {
-            storage.bucket(bucket).delete(done);
-          };
-        }),
-        done
+    after(async () => {
+      Promise.all(
```
Comment (JustinBeckwith) [inline system-test/storage.ts]: Missing the `await`

File: system-test/storage.ts [hunk 4]
```diff
@@ -1179,44 +1103,29 @@ describe('storage', () => {
...
-          );
-          done();
-        }
+      await bucket.addLifecycleRule({
+        action: 'delete',
+        condition: {
+          age: 30,
+          isLive: true,
+        },
+      }),
```
Comment (JustinBeckwith) [inline system-test/storage.ts]: weird formatting - would expect a semicolon here

File: system-test/storage.ts [hunk 5]
```diff
@@ -1235,118 +1144,72 @@ describe('storage', () => {
...
-            if (!err) {
-              done(new Error('Expected an error.'));
-              return;
-            }
-            assert.strictEqual((err as ApiError).code, 403);
-            done();
-          }
-        );
+        await bucket.lock(bucket.metadata.metageneration);
+        try {
```
Comment (JustinBeckwith) [inline system-test/storage.ts]: Please use an `assert.rejects` here so we're 100% sure the exception was thrown

File: system-test/storage.ts [hunk 6]
```diff
@@ -1433,19 +1274,14 @@ describe('storage', () => {
...
-            });
-          },
-          callback
+      async function deleteFilesAsync() {
+        await new Promise(resolve =>
+          setTimeout(resolve, RETENTION_PERIOD_SECONDS * 1000)
+        );
+        return Promise.all(
+          FILES.map(file =>
+            file.setMetadata({temporaryHold: null}).then(() => file.delete())
```
Comment (JustinBeckwith) [inline system-test/storage.ts]: The mix of promises and async/await is weird.  Can we just use an async function here and awaits?

File: system-test/storage.ts [hunk 7]
```diff
@@ -2758,19 +2502,11 @@ describe('storage', () => {
...
-          assert.ifError(err);
-          async.parallel(
-            [file!.delete.bind(file), copiedFile!.delete.bind(copiedFile)],
-            done
-          );
-        });
-      });
+      const [file] = await bucket.upload(FILES.logo.path, opts);
+      const [copiedFile] = await file!.copy('CloudLogoCopy');
+      await Promise.all([file!.delete, copiedFile!.delete()]);
```
Comment (JustinBeckwith) [inline system-test/storage.ts]: Should this be `file.delete()`?

File: system-test/storage.ts (hunk 1 above)
Comment (AVaksman) [reply system-test/storage.ts]: Fixed, thanks

File: system-test/storage.ts (hunk 2 above)
Comment (AVaksman) [reply system-test/storage.ts]: Fixed as well, thanks

File: system-test/storage.ts (hunk 3 above)
Comment (AVaksman) [reply system-test/storage.ts]: added await, thanks

File: system-test/storage.ts (hunk 4 above)
Comment (AVaksman) [reply system-test/storage.ts]: True, fixed it

File: system-test/storage.ts (hunk 5 above)
Comment (AVaksman) [reply system-test/storage.ts]: `assert.fejects` doesn't accept `async` function untill node 10.
reverted to 
```ts
bucket.setRetentionPeriod(RETENTION_DURATION_SECONDS / 2)
      .then(util.noop, err => assert.strictEqual(err.code, 403));
```
What version of Node are we supporting in `system-test`?

File: system-test/storage.ts (hunk 6 above)
Comment (AVaksman) [reply system-test/storage.ts]: Changed that, thanks

File: system-test/storage.ts (hunk 7 above)
Comment (AVaksman) [reply system-test/storage.ts]: Yes it should, fixed, thanks

**Comment by AVaksman** [comment]
> This is such a huge improvement. Thank you!

Thank you.

**Comment by AVaksman** [comment]
@JustinBeckwith PTAL

**Pull request merged**